boot/bootutil/src/loader.c

```diff
@@ -1549,89 +1559,432 @@ boot_set_image_ok(void)
 #endif /* !MCUBOOT_OVERWRITE_ONLY */
 
 /**
- * Performs an image swap if one is required.
+ * Sets the active image on which the bootloader will continue to work.
+ *
+ * @param image_id                  The index of the image (from 0).
+ */
+static void
+boot_set_active_image(size_t image_id)
+{
+    current_image = image_id;
+    boot_data = &boot_data_array[image_id];
+
+#ifdef MCUBOOT_ENC_IMAGES
+    /* The keys used for encryption are no longer valid (may belong to
+     * another image).
+     */
+    boot_enc_zeroize();
+#endif
+}
+
+/**
+ * Check the image dependency whether it is satisfied and modify
+ * the swap type if necessary.
+ *
+ * @param dep               Image dependency which has to be verified.
+ *
+ * @return                  0 on success; nonzero on failure.
+ */
+#if defined(MCUBOOT_OVERWRITE_ONLY) && (BOOT_IMAGE_NUMBER > 1)
+static int
+boot_verify_single_dependency(struct image_dependency *dep)
+{
+    struct image_version *dep_version;
+    size_t dep_slot;
+    int rc;
+
+    /* Determine the source of the image which is the subject of
+     * the dependency and get it's version. */
+    dep_slot = (boot_data_array[dep->image_id].swap_type !=
+                BOOT_SWAP_TYPE_NONE) ?
+                BOOT_SECONDARY_SLOT : BOOT_PRIMARY_SLOT;
+    dep_version = boot_img_version(&boot_data_array[dep->image_id],
+                                   dep_slot);
+
+    rc = boot_is_version_sufficient(&dep->image_min_version, dep_version);
+    if (rc != 0) {
+        /* Dependency not satisfied.
+         * Modify the swap type to decrease the version number of the image
```

Comment:
Using `swap` here seems a bit confusing since this function is currently only enable for overwrite-only mode.

Reply:
It has been enabled for swapping mode too.